lib/net/http/response.gb

```diff
@@ -3,6 +3,25 @@ module Net
     class Response
       attr_accessor :body, :status, :status_code, :protocol, :transfer_encoding, :http_version, :request_http_version, :request
       attr_reader :headers
+
+      def initialize(headers = {})
+        @headers = headers
+      end
+
+      def set_header(key, value)
+        if @headers.is_nil
```

Comment:
We should check this while initializing Response object, not in every `set_header` call